src/fidesops/service/connectors/query_config.py

```diff
@@ -282,23 +289,44 @@ def generate_query(
 
         if filtered_data:
             clauses = []
-            query_data: Dict[str, Tuple[Any, ...]] = {}
+            query_data: List[Dict[str, Any]] = []
             formatted_fields = self.format_fields_for_query(self.fields)
             field_list = ",".join(formatted_fields)
+            from sqlalchemy import bindparam
+
             for field_name, data in filtered_data.items():
                 data = set(data)
                 if len(data) == 1:
                     clauses.append(self.format_clause_for_query(field_name, "="))
-                    query_data[field_name] = (data.pop(),)
+                    # SQL Alchemy + SQLServer errors when we use a tuple in "value"
+                    query_data.append({"name": field_name, "value": data.pop()})
                 elif len(data) > 1:
-                    clauses.append(self.format_clause_for_query(field_name, "IN"))
-                    query_data[field_name] = tuple(data)
+                    # converts to list so that we can generate indexes
+                    newdata = list(data)
+                    query_data_names: List[str] = []
+                    for i in newdata:
+                        # appending "_in_stmt_generated_" (can be any arbitrary str) so that this name has less change of conflicting with pre-existing column in table
+                        query_data_name = "_in_stmt_generated_" + str(newdata.index(i))
+                        query_data.append({"name": query_data_name, "value": i})
+                        query_data_names.append(":" + query_data_name)
+                    # SQL Alchemy + SQLServer errors when the query_str for IN statements was structured like this:
+                    # SELECT order_id,product_id,quantity FROM order_item WHERE order_id IN :some-params-in-tuple
+                    # Current structure of query_str:
+                    # SELECT order_id,product_id,quantity FROM order_item WHERE order_id IN (:_in_stmt_generated_0, :_in_stmt_generated_1, :_in_stmt_generated_2)
+                    clauses.append(
+                        self.format_clause_for_query_in(field_name, query_data_names)
+                    )
```

Comment:
All this custom query generation logic could also be moved to the new `MicrosoftSQLServerQueryConfig`, which is the only place it's required

Reply:
See my comment here- https://github.com/ethyca/fidesops/pull/157/files/35c47ce05d225c529754494d500602f29dd9615b#r786866514 

If we can use the same way to build queries for all connectors, I personally would prefer that, since it'll be less code to maintain

Reply:
My general hang-up with this is that you've added to the `generate_query` complexity quite a bit, also making it diverge even more from the Mongo implementation too.  I'm against bloating the `SQLQueryConfig` class with a bunch of handling necessary for single datastores only.  Your approach doesn't necessarily mean "less code to maintain" - it means more datastores need to be scrutinized on every edit, especially because I'm not confident of our test coverage across datastores. 

But yes, it can be useful to better generalize on the parent class, but watch the complexity tradeoff.

Reply:
Hmm yeah good point. I'm open to either, but wanted to make sure it was clear that this isn't necessarily SQLServer-specific, but instead a different (and more complex!) way to build queries such that the query generation step works for all connectors. I can make a new SQLServer-specific class, and just make a note that it can work across connectors, but more thorough testing will be needed, particularly once we have a better testing framework in place